soh/soh/Enhancements/randomizer/3drando/logic.cpp

```diff
@@ -586,7 +586,7 @@ namespace Logic {
     CanOpenStormGrotto  = CanPlay(SongOfStorms) && (ShardOfAgony || LogicGrottosWithoutAgony);
     HookshotOrBoomerang = CanUse(HOOKSHOT) || CanUse(BOOMERANG);
     CanGetNightTimeGS   = (CanPlay(SunsSong) || !NightGSExpectSuns);
-    CanBreakUpperBeehives = HookshotOrBoomerang || (LogicBombchuBeehives && BombchusInLogic && HasBombchus);
+    CanBreakUpperBeehives = HookshotOrBoomerang || (LogicBombchuBeehives && (BombchusInLogic ? HasBombchus : Bombs));
```

Comment:
What was the reasoning for this change? I thought bombs shouldn't be able to break the upper beehives anyway, and it wouldn't make sense for them to only be relevant if bombchus aren't in logic

Reply:
https://github.com/HarbourMasters/Shipwright/pull/3144#discussion_r1416260945

Not sure if this is how you're supposed to handle this situation, but spirit trial (the obvious bombchu requirement) seems to be in a weird state involving bombchu logic too, as it only uses HasBombchus, so I'm not sure if copying that is wise.

Reply:
After discussion, we decided this was a bombchu problem out of scope for this PR, so HasBombchus should be trusted for the time being.